src/EventLoop.hs

```diff
@@ -122,59 +122,57 @@ runGithubEventLoop ghQueue enqueueEvent = runLoop
       runLoop
 
 runLogicEventLoop
-  :: MonadIO m
-  => MonadLogger m
-  => MonadMonitor m
+  ::  forall es
+  .  IOE :> es
+  => Metrics.MetricsOperation :> es
+  => Time.TimeOperation :> es
+  => Git.GitOperation :> es
+  => GithubApi.GithubOperation :> es
+  => MonadLoggerEffect :> es
   => TriggerConfiguration
   -> ProjectConfiguration
   -> MergeWindowExemptionConfiguration
-  -- Interpreters for Git and GitHub actions.
-  -> (forall a. Metrics.MetricsOperationFree a -> m a)
-  -> (forall a. Time.TimeOperationFree a -> m a)
-  -> (forall a. Git.GitOperationFree a -> m a)
-  -> (forall a. GithubApi.GithubOperationFree a -> m a)
   -- Action that gets the next event from the queue.
-  -> m (Maybe Logic.Event)
+  -> IO (Maybe Logic.Event)
```

Comment:
These functions should probably fit in their own effects. This one would make for a nice `EventQueue` effect, and the `ProjectState` update below would fit in a similarly named effect.

I avoided refactoring further than the rewrite itself in this PR. I rewrote them to be `IO` because that's what they were in all call sites, and there was no `m` anymore.